### ISSUE TITLE
[🐞] server actions can't run inside useClientEffect$ in preview mode

### ISSUE THREAD

**Issue by genki** [BuilderIO/qwik#2798]
### Which component is affected?

Qwik City (routing)

### Describe the bug

**Doing**:
I am using server actions inside `useClientEffect$(() => ...)` like this,

```ts
export const User = action$(signIn);

export default component$(() => {
    ...
    const getUser = User.use();
    useClientEffect$(() => {
      const form = new FormData();
      form.append('userId', userId);
      getUser.run(form).then((res) => ...);
```

 and it works correctly in dev mode, so then I am trying to build it for preview mode.

**Expect**:

It succeeds.

**Happening**:

`npm run build` said:

```
error  Identifier ("getUser") can not be captured inside the scope (useClientEffect$) because "getUser.run" is a function, which is not serializable.
```

It's a bit confusing. Because server actions should run inside `onClick$` so they should be able to go across the `$` sign border.
Am I misunderstanding?

### Reproduction

https://stackblitz.com/edit/qwik-starter-ubxny6?file=src/routes/index.tsx

### Steps to reproduce

Run `npm install && npm run build` in terminal.

### System Info

```shell
System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M2
    Memory: 49.78 MB / 24.00 GB
    Shell: 3.5.1 - /opt/homebrew/bin/fish
  Binaries:
    Node: 19.5.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.3.1 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 109.0.5414.119
    Firefox: 109.0
    Safari: 16.2
  npmPackages:
    @builder.io/qwik: file:../clone/qwik/packages/qwik => 0.17.2 
    @builder.io/qwik-city: file:../clone/qwik/packages/qwik-city => 0.1.0-beta11 
    undici: 5.14.0 => 5.14.0 
    vite: 4.0.3 => 4.0.3
```

**Comment by manucorporat** [comment]
Already opened a PR, but why would you call an action from a `useClientEffect$()` sounds like a bad idea

**Comment by manucorporat** [comment]
Actions should be used as response to user actions, not to get data, look into `loader$()` to get data ASAP

**Comment by genki** [comment]
Thank you for fixing :)

At first, I have tried to use `loader$` for fetching data that takes a time.
But then I found `loader$` stops rendering until the data arrived even if the return value was Promise.
I would like to present empty container with the loading message first, and then the data if arrived.
I am happy if `loader$` can acts as lazily loading.
As you mentioned, using `action$` inside `useClientEffect$` is not good solution but was a workaround for that.

One more try was using `useResource$`.
The problem was `onPending` doesn't work if the fetching process has run in server side.
If there is an option to show loading message while server side data fetching, I would like to use that.
